modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/approval/common/UserApprovalTestBase.java

```diff
@@ -200,50 +202,66 @@ protected void addAssociationForMatch() throws Exception {
+                    log.info("BPEL Workflow deployed successfully in " + count + " attempt(s).");
                 }
             }
 
-            // Wait 20 times for 5 seconds intervals until the workflow is properly deployed. This is for server based
-            // test runners like Jenkins.
-            if (count < 20) {
-                log.info("Still no luck :(. So going to wait 5 seconds and try " + (20 - count) + " more time(s).");
-                Thread.sleep(5000);
+            if(isBpelDeployed) {
+                log.info("Verifying Human Task deployment.");
+                url = identityContextUrls.getSecureServiceUrl() + "/" + addUserWorkflowName + "TaskService?wsdl";
+                if (isWorkflowDeployed(url)) {
+                    isHumanTaskDeployed = true;
+                    log.info("Human Task Workflow deployed successfully in " + count + " attempt(s).");
+                    log.info("Workflow is deployed successfully.");
+                    break;
+                }
             }
+            log.info("Still no luck :(. So going to wait 1 seconds and retry");
+            Thread.sleep(1000);
+        }
+
+        // Give up after 100 seconds.
+        if (!isBpelDeployed || !isHumanTaskDeployed) {
+            log.info("No luck. Going to give up. Test will most probably fail.");
+        }
+        log.info("Waiting for 5 seconds before proceeding.");
+        Thread.sleep(5000);
+    }
+
+    private boolean isWorkflowDeployed(String url) throws IOException {
 
-            // Give up after 100 seconds or this will be a forever loop.
-            if (count >= 20) {
-                log.info("No luck. Going to give up. Test will most probably fail.");
-                runLoop = false;
+        // We have to check whether the service is up and running by calling the generated endpoint.
+        HttpClient client = new HttpClientFactory().getHttpClient();
+        HttpGet request = new HttpGet(url);
+        HttpResponse httpResponse = client.execute(request);
+
+        // If the server response is 500, or it contains "service not available" text or "Operation not found" text,
+        // then we have to assume that the service is still not available.
+        // Log response to get more context.
+        log.info("Response code: " + httpResponse.getStatusLine().getStatusCode());
+        BufferedReader bufferedReader =
+                new BufferedReader(new InputStreamReader(httpResponse.getEntity().getContent()));
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            log.info("Response content: " + line);
+            if (line.contains("The service cannot be found for the endpoint reference") ||
+                    line.contains("The endpoint reference (EPR) for the Operation not found")) {
+                break;
             }
-            count++;
-        } while (runLoop);
+        }
+
+        return (httpResponse.getStatusLine().getStatusCode() == 200);
```

Comment:
shouldn't the status code be checked before and returned? we would not need to read the body if the expected code is not received

Reply:
addressed by af0f52a25c490ea7670ae5b3a8b7628cc45b903a